src/CppUTestExt/MockExpectedCallsList.cpp

```diff
@@ -140,74 +141,116 @@ void MockExpectedCallsList::addExpectations(const MockExpectedCallsList& list)
         addExpectedCall(p->expectedCall_);
 }
 
+// Tracking the parent is needed because we need to know it in order to remove an element
+// during the iteration.
+// parent must not advance if p has been deleted (we achieve this by setting p=parent at the end
+// of pruneEmptyNodeFromList(), and letting the "parent = p" execute normally in the for loop).
+//
+// You need to close the curly bracket after the processing step.
+#define ITERATE_WITH_PREV_AND_NEXT \
```

Comment:
I think keeping the iteration algorithm in one place is good, but I'm ok with inlining this everywhere if you find it ugly.

PREV should be changed to PARENT